x/crosschain/keeper/migrate_tss_funds.go

```diff
@@ -0,0 +1,114 @@
+package keeper
+
+import (
+	"fmt"
+	"sort"
+
+	errorsmod "cosmossdk.io/errors"
+	sdkmath "cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	tmbytes "github.com/tendermint/tendermint/libs/bytes"
+	tmtypes "github.com/tendermint/tendermint/types"
+	"github.com/zeta-chain/zetacore/common"
+	"github.com/zeta-chain/zetacore/x/crosschain/types"
+)
+
+func (k Keeper) MigrateTSSFundsForChain(ctx sdk.Context, chainID int64, amount sdkmath.Uint) error {
+	currentTss, found := k.GetTSS(ctx)
```

Comment:
We're already getting the current TSS in `MigrateTssFunds`, maybe we could have it as a parameter for this function